File: 9.cpp
struct Solution {
    constexpr bool isPalindrome(int x) {
        if (x == 0){
            return true;
        }
        if (x < 0 || x % 10 == 0){
            return false;
        }
        long xx = x, y = 0;
        while (xx > 0){
            y = y * 10 + (xx % 10);
            xx /= 10;
        }
        return x == y;
    }
};

File: sol/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True
        if x < 0 or x % 10 == 0:
            return False
        xx, y = x, 0
        while xx > 0:
            xx, r = divmod(xx, 10)
            y = y * 10 + r
        return x == y
